vspace/src/main/java/edu/asu/diging/vspace/core/services/impl/SpaceManager.java

```diff
@@ -297,4 +304,83 @@ public Iterable<Space> addIncomingLinkInfoToSpaces(Iterable<Space> spaces) {
     public Page<ISpace> findByNameOrDescription(Pageable requestedPage, String searchText) {
         return spaceRepo.findDistinctByNameContainingOrDescriptionContaining(requestedPage, searchText,searchText);
     }
+    
+    @Override
+    public List<ISpace> findByName(String searchText){
+        String searchTerm = "%" + searchText + "%";
+        List<Space> spaces = spaceRepo.findByNameLike(searchTerm);
+        List<ISpace> spaceResults = new ArrayList<>();
+        spaces.forEach(r -> spaceResults.add(r));
+        return spaceResults;
+    }
+    
+    @Override
+    public List<ISpace> getSpaces(int pageNo) {
+        return getSpaces(pageNo, SortByField.CREATION_DATE.getValue(), Sort.Direction.DESC.toString());
+    }
+    
+    /**
+     * Method to return the requested spaces
+     * 
+     * @param pageNo. if pageNo<1, 1st page is returned, if pageNo>total pages,last
+     *                page is returned
+     * @return list of images in the requested pageNo and requested order.
+     */
+
+    @Override
+    public List<ISpace> getSpaces(int pageNo, String sortedBy, String order) {
+        Sort sortingParameters = getSortingParameters(sortedBy, order);
+        pageNo = validatePageNumber(pageNo);
```

Comment:
The two methods `validatePageNumber` and `getTotalPages` only seem to be used from here (so they should be private) but also I'm not sure if you even need them. I think if you try to get a page number that's higher than the number of pages it simply returns no result. But the page object you get back has the number of pages in it, so in that case you can simply make another call to the db. This way in most cases you only make one DB call except when `pageNo` is greater than max pages. The way it's implemented right now there are always two db calls.